### PR TITLE
CA-247399: New pool join rules to reflect ely and beyond changes in t…

### PR DESCRIPTION
**Pull request by kc284**
…he updates.

Removed comparison of host patches and implemented comparison on the api version
and database schema instead.

Signed-off-by: Konstantina Chremmou <konstantina.chremmou@citrix.com>

### CODE REVIEW

**Comment by lindig** [comment]
This looks good to me. Could you edit the commit message such that the headline is self-contained does not spill into the body?

File: ocaml/xapi/xapi_pool.ml [hunk 1]
```diff
@@ -111,29 +111,33 @@ let pre_join_checks ~__context ~rpc ~session_id ~force =
...
+  in
+
+  let assert_db_schema_matches () =
+    let master = get_master rpc session_id in
+    let candidate_slave = Helpers.get_localhost ~__context in
+    let master_sw_version = Client.Host.get_software_version ~rpc ~session_id ~self:master in
+    let slave_sw_version = Db.Host.get_software_version ~__context ~self:candidate_slave in
+    let master_db_schema = try List.assoc Xapi_globs._db_schema master_sw_version with _ -> "" in
+    let slave_db_schema = try List.assoc Xapi_globs._db_schema slave_sw_version with _ -> "" in
+    if master_db_schema <> "" && slave_db_schema <> "" && master_db_schema <> slave_db_schema then
```
Comment (lindig) [inline ocaml/xapi/xapi_pool.ml]: When an error is raised while retrieving `*_db_schema`, this leads to empty strings and no check. Could this lead to problems later on? Would it not be safer to raise an error?

File: ocaml/idl/datamodel.ml [hunk 2]
```diff
@@ -906,6 +906,10 @@ let _ =
     ~doc:"The pool failed to disable the external authentication of at least one host." ();
   error Api_errors.pool_auth_disable_failed_permission_denied ["host";"message"]
     ~doc:"The pool failed to disable the external authentication of at least one host." ();
+  error Api_errors.pool_joining_host_must_have_same_api_version []
```
Comment (robhoes) [inline ocaml/idl/datamodel.ml]: How about adding the master and joining-host's versions as arguments to the error? Similarly below.

File: ocaml/xapi/xapi_pool.ml (hunk 1 above)
Comment (robhoes) [inline ocaml/xapi/xapi_pool.ml]: I think that it is better to raise an error.

**Pull request merged**